### PR TITLE
Add opt-in notebook execution info status bar

### PR DESCRIPTION
**Pull request by seeM**
Addresses #5353.

### QA Notes

Enabling the `notebook.experimental.showExecutionInfo` setting should show the last execution info in the status bar on the bottom right of the Positron window.

Tests should also pass locally.

### CODE REVIEW

File: extensions/positron-notebook-controllers/test-workspace/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (seeM) [inline extensions/positron-notebook-controllers/test-workspace/.vscode/settings.json:1]: An empty workspace is needed to update the config during tests.

File: extensions/positron-notebook-controllers/src/notebookController.ts [hunk 2]
```diff
@@ -18,6 +18,21 @@ enum NotebookCellOutputType {
...
+}
+
+/** An event which fires when a notebook execution ends. */
+export interface DidEndExecutionEvent {
+	/** The notebook cells that were executed. */
+	cells: vscode.NotebookCell[];
+
+	/** The duration of the execution in milliseconds. */
+	duration: number;
+}
```
Comment (nstrayer) [inline extensions/positron-notebook-controllers/src/notebookController.ts:34]: Would we potentially want to add the execution end type? E.g. `'success' | 'error' | 'canceled'`?

File: extensions/positron-notebook-controllers/src/statusBar.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	// Update the item's text when an execution starts.
+	disposables.push(manager.onDidStartExecution((e) => {
+		item.text = `Executing ${formatCount(e.cells.length, 'cell')}`;
+	}));
+
+	// Update the item's text when an execution ends.
+	disposables.push(manager.onDidEndExecution((e) => {
+		const durationSeconds = Math.round(e.duration / 1000);
+		item.text = `Executed ${formatCount(e.cells.length, 'cell')} ` +
+			`in ${formatCount(durationSeconds, 'second')}`;
```
Comment (nstrayer) [inline extensions/positron-notebook-controllers/src/statusBar.ts:37]: Will we potentially want more precision than the second level?

File: extensions/positron-notebook-controllers/test-workspace/.vscode/settings.json (hunk 1 above)
Comment (nstrayer) [reply extensions/positron-notebook-controllers/test-workspace/.vscode/settings.json:1]: another curse of not having json comments.

File: extensions/positron-notebook-controllers/src/statusBar.ts (hunk 3 above)
Comment (nstrayer) [reply extensions/positron-notebook-controllers/src/statusBar.ts:37]: This is how they format time stamps. Not sure how fancy we need to get for testing though. Maybe @testlabauto  has an opinion?
https://github.com/posit-dev/positron/blob/25f4eaedb24cc555ff60683beffe465c32ae968a/src/vs/workbench/contrib/notebook/browser/contrib/cellStatusBar/executionStatusBarItemController.ts#L23-L37

**Review by nstrayer — DISMISSED** [review]
Looks good and works for me! Just a few comments.

File: extensions/positron-notebook-controllers/package.nls.json [hunk 4]
```diff
@@ -1,5 +1,6 @@
 {
 	"positron.command.restartKernel.title": "Restart Kernel",
 	"positron.command.restartKernel.shortTitle": "Restart",
-	"python.languageServer.description": "This setting is included as a temporary workaround to disable the Jupyter extension's language server. The value should remain 'Default'."
+	"python.languageServer.description": "This setting is included as a temporary workaround to disable the Jupyter extension's language server. The value should remain 'Default'.",
+	"notebook.experimental.showExecutionInfo.description": "Enable a status bar item that shows notebook execution info."
```
[the remark this answers is not in the document]
Comment (seeM) [reply extensions/positron-notebook-controllers/package.nls.json]: Good point, added in 6d2c1d7

File: extensions/positron-notebook-controllers/src/statusBar.ts (hunk 3 above)
Comment (seeM) [reply extensions/positron-notebook-controllers/src/statusBar.ts:37]: We may want these sorts of refinements/additions in future. I opted for something super simple for now that matches Jupyter Lab's behavior enough to satisfy @testlabauto's benchmarking needs.

**Pull request merged**